library/Opus/Search/Solr/Document/Base.php

```diff
@@ -0,0 +1,164 @@
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Application
+ * @author      Thomas Urban <thomas.urban@cepharum.de>
+ * @copyright   Copyright (c) 2009-2015, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ * @version     $Id$
+ */
+
+abstract class Opus_Search_Solr_Document_Base {
+
+	public function __construct( Zend_Config $options ) {}
+
+	/**
+	 * Retrieves XML describing model data of provided Opus document.
+	 *
+	 * @param Opus_Document $opusDoc
+	 * @return DOMDocument
+	 */
+	protected function getModelXml( Opus_Document $opusDoc ) {
+		// Set up caching xml-model and get XML representation of document.
+		$caching_xml_model = new Opus_Model_Xml;
+		$caching_xml_model->setModel( $opusDoc );
+		$caching_xml_model->excludeEmptyFields();
+		$caching_xml_model->setStrategy( new Opus_Model_Xml_Version1 );
+		$cache = new Opus_Model_Xml_Cache( $opusDoc->hasPlugin( 'Opus_Document_Plugin_Index' ) );
+		$caching_xml_model->setXmlCache( $cache );
+
+		$modelXml = $caching_xml_model->getDomDocument();
+
+		// extract fulltext from file and append it to the generated xml.
+		$this->attachFulltextToXml( $modelXml, $opusDoc->getFile(), $opusDoc->getId() );
+
+		return $modelXml;
+	}
+
+	/**
+	 * Appends fulltext data of every listen file to provided XML document.
+	 *
+	 * @param DomDocument $modelXml
+	 * @param Opus_File[] $files
+	 * @param string $docId ID of document
+	 * @return void
+	 */
+	private function attachFulltextToXml( $modelXml, $files, $docId ) {
+
+		// get root element of XML document containing document's information
+		$docXml = $modelXml->getElementsByTagName( 'Opus_Document' )->item( 0 );
+		if ( is_null( $docXml ) ) {
+			Opus_Log::get()->warn( 'An error occurred while attaching fulltext information to the xml for document with id ' . $docId );
+			return;
+		}
+
+
+		// only consider files which are visible in frontdoor
+		$files = array_filter( $files, function ( $file ) {
+			/** @var Opus_File $file */
+			return $file->getVisibleInFrontdoor() === '1';
+		} );
+
+		if ( !count( $files ) ) {
+			// any attached file is hidden from public
+			$docXml->appendChild( $modelXml->createElement( 'Has_Fulltext', 'false' ) );
+			return;
+		}
+
+		$docXml->appendChild( $modelXml->createElement( 'Has_Fulltext', 'true' ) );
+
+
+		// fetch reference on probably separate service for extracting fulltext data
+		$extractingService = Opus_Search_Service::selectExtractingService();
+
+		// extract fulltext data for every file left in set after filtering before
+		foreach ( $files as $file ) {
+			$fulltext = '';
+
+			try {
+				$fulltext = $extractingService->extractDocumentFile( $file );
+				$fulltext = trim( iconv( "UTF-8", "UTF-8//IGNORE", $fulltext ) );
+			} catch ( Opus_Search_Exception $e ) {
```

Comment:
Okay, this is fixing issues on SolrIndexBuilder, isn't it!? Having just pushed commit including that extra catch.